### PR TITLE
Replace `past` with `past_key_values`

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
The argument `past` was completely replaced with `past_key_values` thus this PR should fix any problem with `kwargs` being swallowed for old models in generation. 
Related to #20347

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for working on this! There seems to be an issue with the TensorFlow tests. Also would like to have @gante opinion on this before merging.

**Review by gante — APPROVED** [review]
LGTM 👍 

Although the change in the failing TF test is weird, we should try to understand it before we merge this PR 🤔

File: src/transformers/models/bart/modeling_tf_bart.py [hunk 1]
```diff
@@ -1475,6 +1475,16 @@ def prepare_inputs_for_generation(
...
+    @staticmethod
+    def _reorder_cache(past, beam_idx):
+        reordered_past = ()
+        for layer_past in past:
+            # cached cross_attention states don't have to be reordered -> they are always the same
+            reordered_past += (
+                tuple(tf.gather(past_state, beam_idx, axis=0) for past_state in layer_past[:2]) + layer_past[2:],
+            )
+        return reordered_past
+
```
[the remark this answers is not in the document]
Comment (ArthurZucker) [reply src/transformers/models/bart/modeling_tf_bart.py:1487]: Sure, I thought that the failing test might come from this but I was wrong 😉

**Comment by ArthurZucker** [comment]
Ok the failing tests were because I did not pull from main, were the `tf_utils` now uses the `generate_config`. LGTM the failing test seems to be unrelated

**Comment by sgugger** [comment]
Yes, good to merge for me!

**Pull request merged**